include/aspect/parameters.h

```diff
@@ -202,6 +202,109 @@ namespace aspect
      */
 
     /**
+     * @name Formulation settings
+     * @{
+     */
+    struct CombinedFormulation
+    {
+      enum Kind
+      {
+        BA,
+        ALA,
+        isothermal_compression,
+        custom
+      };
+      static Kind parse(const std::string &input)
+      {
+        if (input == "isothermal compression")
+          return CombinedFormulation::isothermal_compression;
+        else if (input == "ALA")
+          return CombinedFormulation::ALA;
+        else if (input == "BA")
+          return CombinedFormulation::BA;
+        else if (input == "custom")
+          return CombinedFormulation::custom;
+        else
+          AssertThrow(false, ExcNotImplemented());
+        return CombinedFormulation::isothermal_compression;
+      }
+    };
+
+    struct FormulationMassConservation
+    {
+      enum Kind
+      {
+        isothermal_compression,
+        reference_density_profile,
+        implicit_reference_density_profile,
+        incompressible,
+        ask_material_model
```

Comment:
I think that from the use later on, you use `ask_material_model` as a "yet-to-be-determined" value, right? Maybe call it that way.